feature/gribi/ate_tests/route_removal_non_default_vrf_test/route_removal_non_default_vrf_test.go

```diff
@@ -0,0 +1,454 @@
+	"testing"
+	"time"
+
+	"github.com/openconfig/featureprofiles/internal/attrs"
+	"github.com/openconfig/featureprofiles/internal/deviations"
+	"github.com/openconfig/featureprofiles/internal/fptest"
+	"github.com/openconfig/featureprofiles/internal/gribi"
+	"github.com/openconfig/gribigo/fluent"
+	"github.com/openconfig/ondatra"
+	"github.com/openconfig/ondatra/telemetry"
+	"github.com/openconfig/ondatra/telemetry/ateflow"
+	"github.com/openconfig/ygot/ygot"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"google.golang.org/protobuf/encoding/prototext"
+	"google.golang.org/protobuf/proto"
+
+	gpb "github.com/openconfig/gribi/v1/proto/service"
+)
+
+func TestMain(m *testing.M) {
+	fptest.RunTests(m)
+}
+
+// Settings for configuring the baseline testbed with the test
+// topology.
+//
+// The testbed consists of ate:port1 -> dut:port1,
+// dut:port2 -> ate:port2.
+//
+//   * ate:port1 -> dut:port1 subnet 192.0.2.0/30
+//   * ate:port2 -> dut:port2 subnet 192.0.2.4/30
+//
+//   * Destination network for traffic test: 198.51.100.0/24
+
+const (
+	ateDstNetEntryNonDefault = "198.51.100.0/24" // IP Entry to be injected in non-default network instance
+	ateDstNetEntryDefault    = "203.0.113.0/24"  // IP Entry to be injected in default network instance
+	nonDefaultVRF            = "VRF-1"           // Name of non-default network instance
+	clientAOriginElectionID  = 10
+	clientBOriginElectionID  = 9
+	nhIndex                  = 1
+	nhgIndex                 = 42
+)
+
+var (
+	dutPort1 = attrs.Attributes{
+		Desc:    "dutPort1",
+		IPv4:    "192.0.2.1",
+		IPv4Len: 30,
+	}
+
+	atePort1 = attrs.Attributes{
+		Name:    "atePort1",
+		IPv4:    "192.0.2.2",
+		IPv4Len: 30,
+	}
+
+	dutPort2 = attrs.Attributes{
+		Desc:    "dutPort2",
+		IPv4:    "192.0.2.5",
+		IPv4Len: 30,
+	}
+
+	atePort2 = attrs.Attributes{
+		Name:    "atePort2",
+		IPv4:    "192.0.2.6",
+		IPv4Len: 30,
+	}
+)
+
+// TestRouteRemovalNonDefaultVRFFlush test flush with the following operations
+// 1. Flush request from clientA (the primary client) non default VRF should succeed.
+// 2. Flush request from clientB (not a primary client) non default VRF should fail.
+// 3. Failover the primary role from clientA to clientB.
+// 4. Flush from clientB non default VRF should succeed.
+// 5. Flush from default VRF should not delete all the gRIBI objects like NH and NHG.
+func TestRouteRemovalNonDefaultVRFFlush(t *testing.T) {
+	ctx := context.Background()
+
+	dut := ondatra.DUT(t, "dut")
+	configureDUT(t, dut)
+
+	ate := ondatra.ATE(t, "ate")
+	ateTop := configureATE(t, ate)
+
+	ateTop.Push(t).StartProtocols(t)
+	configureNetworkInstance(t, dut)
+	configureDUT(t, dut)
+
+	// Configure the gRIBI client clientA with election ID of 10.
+	clientA := &gribi.Client{
+		DUT:                  dut,
+		Persistence:          true,
+		InitialElectionIDLow: clientAOriginElectionID,
+	}
+
+	defer clientA.Close(t)
+
+	t.Log("Establish gRIBI clientA connection with PERSISTENCE set to TRUE")
```

Comment:
optional: Full line sentences typically capitalized and with ending punctuation.
A uniformed style is appreciated (e.g. line 142)
https://github.com/golang/go/wiki/CodeReviewComments#error-strings